controller/node_controller.go

```diff
@@ -1356,7 +1356,33 @@ func (nc *NodeController) getNewAndMissingOrphanedReplicaDataStores(diskName, di
 	return newOrphanedReplicaDataStores, missingOrphanedReplicaDataStores
 }
 
-func (nc *NodeController) deleteOrphans(node *longhorn.Node, diskName string, diskInfo *monitor.CollectedDiskInfo, missingOrphanedReplicaDataStores map[string]string) error {
+func (nc *NodeController) deleteInstanceOrphans(node *longhorn.Node) error {
+	if !node.Spec.EvictionRequested {
+		return nil
+	}
```

Comment:
Move out of the function.
This will be better

```
if node.Spec.EvictionRequested {
    err = deleteOrphansForEngineAndReplicaInstances()
     ....
}
```